### PR TITLE
BGDIINF_SB-1385: Corrected serializer validation & Implemented Collection de-serialization

### PR DESCRIPTION
**Pull request by ltshb**
In a model serializer when explicitely declaring the fields, we need to
add the validation manually and these validations should then match the
one from the models.
Note that when the fields are not manually declared but simply listed in
the Meta.fields then the validation is automatically generated with the
validator from the models.

Also finalize the colleciton de-serialization with unit tests

### CODE REVIEW

File: app/stac_api/serializers.py [hunk 1]
```diff
@@ -238,19 +260,24 @@ class Meta:
...
-    crs = serializers.SerializerMethodField()
+    # NOTE: when explicitely declaring fields, we need to had the validation as for the field
+    # in model !
+    id = serializers.CharField(
+        required=True,
+        max_length=255,
+        source="name",
+        validators=[validate_name, UniqueValidator(queryset=Collection.objects.all())]
+    )
+    links = CollectionLinkSerializer(required=False, many=True)
```
Comment (hansmannj) [inline app/stac_api/serializers.py:271]: according to specs, links are required (at least `href` and `rel` are required)

File: app/stac_api/serializers.py [hunk 2]
```diff
@@ -416,6 +491,27 @@ class Meta:
...
         ]
 
+    # NOTE: when explicitely declaring fields, we need to had the validation as for the field
+    # in model !
+    type = serializers.CharField(source='media_type', max_length=200)
+    # Here we need to explicitely define these fields with the source, because they are renamed
+    # in the get_fields() method
+    eo_gsd = serializers.FloatField(source='eo_gsd', required=False, allow_null=True)
+    geoadmin_lang = serializers.ChoiceField(
+        source='geoadmin_lang', choices=['de', 'fr', 'it', 'rm', 'en']
```
Comment (hansmannj) [inline app/stac_api/serializers.py:501]: probably an `allow_blank = True` is missing here, as in the Language class "NONE" (i.e. "") is a valid option.

File: app/tests/test_serializer.py [hunk 3]
```diff
@@ -128,12 +129,191 @@ def test_collection_serialization(self):
...
         self.check_stac_collection(expected, python_native)
 
-        # back-transate to Python native:
-        stream = io.BytesIO(content)
-        data = JSONParser().parse(stream)
-        # back-translate into fully populated collection instance:
-        serializer = CollectionSerializer(data=data, context={'request': request})
-        self.assertEqual(True, serializer.is_valid(), msg='Serializer data not valid.')
+    def test_collection_deserialization_create_only_required(self):
+        data = OrderedDict([
```
Comment (hansmannj) [inline app/tests/test_serializer.py:133]: according to specs `links` should be in the required minimum, too.

**Review by hansmannj — CHANGES_REQUESTED** [review]
Nice solutions and good test coverage :smile: :+1:
Just some minor comments.

File: app/stac_api/serializers.py [hunk 4]
```diff
@@ -213,7 +229,13 @@ class CollectionLinkSerializer(NonNullModelSerializer):
 
     class Meta:
         model = CollectionLink
-        fields = ['href', 'rel', 'link_type', 'title']
+        fields = ['href', 'rel', 'title', 'type']
+
+    # NOTE: when explicitely declaring fields, we need to had the validation as for the field
```
[the remark this answers is not in the document]
Comment (ltshb) [reply app/stac_api/serializers.py:234]: Copy paste of typo

File: app/stac_api/serializers.py (hunk 1 above)
Comment (ltshb) [reply app/stac_api/serializers.py:271]: Yes but, links are automatically generated, here we provide the ability to add extra non-generated links but these ones are not required. I'll add a comment though.

File: app/tests/test_serializer.py (hunk 3 above)
Comment (ltshb) [reply app/tests/test_serializer.py:133]: Same as above links are generated (self, root, parent, ...)

**Review by rebert — APPROVED** [review]
Nice - lgtm

File: app/stac_api/serializers.py (hunk 2 above)
Comment (ltshb) [reply app/stac_api/serializers.py:501]: Good point, I'll add the `required=False` and `allow_blank=True` to match the SPEC and add a comment in https://jira.swisstopo.ch/browse/BGDIINF_SB-1372 which will fully implement the asset deserializer

**Review by hansmannj — APPROVED** [review]
:+1:

**Pull request merged**